core/dbt/parser/sources.py

```diff
@@ -344,11 +345,32 @@ def get_unused_msg(
         return '\n'.join(msg)
 
 
+def merge_freshness_time_thresholds(
+    base: Optional[Time], update: Optional[Time]
+) -> Optional[Time]:
+    if base is not None and update is not None:
+        return base.merged(update)
+    elif base is None and update is not None:
+        return update
+    elif base is not None and update is None:
+        return base
+    else:
+        # base and update are none
+        return None
```

Comment:
Here's a more terse way to express this function that leans more into EAFP:

```suggestion
    try:
        return base.merged(update)
    except AttributeError:
        return update or base
```